pandas/_libs/tslibs/parsing.pyx

```diff
@@ -531,21 +534,83 @@ def try_parse_datetime_components(object[:] years,
 # ----------------------------------------------------------------------
 # Miscellaneous
 
-_DATEUTIL_LEXER_SPLIT = None
-try:
-    # Since these are private methods from dateutil, it is safely imported
-    # here so in case this interface changes, pandas will just fallback
-    # to not using the functionality
-    from dateutil.parser import _timelex
-
-    if hasattr(_timelex, 'split'):
-        def _lexer_split_from_str(dt_str):
-            # The StringIO(str(_)) is for dateutil 2.2 compatibility
-            return _timelex.split(StringIO(str(dt_str)))
 
-        _DATEUTIL_LEXER_SPLIT = _lexer_split_from_str
-except (ImportError, AttributeError):
-    pass
+# Class copied verbatim from https://github.com/dateutil/dateutil/pull/732
+#
+# Copyright (c) 2017 - dateutil contributors
+class _timelex(object):
```

Comment:
It occurs to me - does this need to be a Python class or can it be a `cdef` class?

Maybe you want to do it in two steps, but there are probably a few other ways where this can be Cython-ed up.

Reply:
We probably could do some Cython optimizations, but I think it's best we get the porting done first using the code *as is*.

Ideally, we wouldn't even have this class - we would just import it from outside 😉.